### PR TITLE
Fix/modal

### PR DESCRIPTION
**Pull request by jakobe**
This PR fixes 2 modal bugs that was introduced with #988:
* Respecting iOS safe area in modal flavor (and prevent it in drawer flavor as drawer already has top padding)
* Jagged close animation when keyboard is visible because of height of viewport:
When closing modal (call `.close()`, click close button or swiping header) we'll blur active element to force keyboard to close (if keyboard is visible we add a small delay to allow keyboard to hide before starting close animation)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 1]
```diff
@@ -64,9 +65,28 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     this.ionContent.scrollToBottom(scrollDuration || 0);
   }
 
-  close(data?: any) {
+  async close(data?: any): Promise<void> {
```
Comment (lydemann) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:68]: No tests for this?

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 2]
```diff
@@ -86,6 +106,22 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     }
   }
 
+  @HostListener('window:keyboardWillShow')
```
Comment (lydemann) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:109]: Tests?

Comment (askarby) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:109]: I believe that the `keyboardWillShow`-event is dispatched by cordova (capacitor), but won't we experience "the same wierdness" when running on a mobile browser?

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 3]
```diff
@@ -86,6 +106,22 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
...
+  _onKeyboardWillShow() {
+    this.keyboardVisible = true;
+  }
+
+  @HostListener('window:keyboardWillHide')
```
Comment (askarby) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:114]: I believe that the `keyboardWillHide`-event is dispatched by cordova (capacitor), but won't we experience "the same wierdness" when running on a mobile browser?

Comment (jakobe) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:114]: @askarby Yep, this is a "temporary" fix for the native app - we are nowhere near supporting PWAs on native devices yet...

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts (hunk 2 above)
Comment (lydemann) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:109]: In the case, that this is also a problem for mobile screen keyboards, it seems that the resize event will be an option:
https://stackoverflow.com/questions/27132796/is-there-any-javascript-event-fired-when-the-on-screen-keyboard-on-mobile-safari

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts (hunk 1 above)
Comment (jakobe) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:68]: Added 👌

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts (hunk 2 above)
Comment (jakobe) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:109]: Added 👌

**Pull request merged**